lib/Workflow/Condition.pm

```diff
@@ -61,14 +62,51 @@ sub evaluate_condition {
         $condition = $wf->_factory()
             ->get_condition( $orig_condition, $wf->type );
         $log->debug( "Evaluating condition '$orig_condition'" );
-        my $return_value = $condition->evaluate($wf);
+
+        my $return_value;
+        my $result = $condition->evaluate($wf);
+        if (ref $result eq 'Workflow::Condition::IsTrue') {
+            $log->info( "Got true result with '$result' on '$orig_condition'");
+            $return_value = 1;
+        } elsif (ref $result eq 'Workflow::Condition::IsFalse') {
```

Comment:
actually, the above code completely works even when an instance of `Workflow::Condition::IsFalse` evaluates to false in boolean expressions: you're not checking the actual value of `$result`.